datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -454,141 +587,6 @@ fn build_row_group_predicate(
     )
 }
 
-fn read_partition_no_file_columns(
```

Comment:
This logic is moved and reformulated into the Iterator implementation for [`ParquetExecStream`]. This simplifies the code and also allows it to be used without an intermediate buffer